libs/sdk-model/src/execution/filter/factory.ts

```diff
@@ -249,3 +265,34 @@ export function newMeasureValueFilter(
         };
     }
 }
+
+/**
+ * Creates a new ranking filter.
+ *
+ * @param measureOrRef - instance of measure to filter, or reference of the measure object; if instance of measure is
+ *   provided, then it is assumed this measure is in scope of execution and will be referenced by the filter by
+ *   its local identifier
+ * @param operator - TOP or BOTTOM operator to use in the filter
+ * @param value - TBD
+ * @param attributesOrRefs - Optional. Array of attributes used in filter, or reference of the attribute object. If instance of attribute is
+ *   provided, then it is assumed this attribute is in scope of execution and will be referenced by the filter by
+ *   its local identifier
+ * @public
+ */
+export function newRankingFilter(
+    measureOrRef: IMeasure | ObjRefInScope | string,
+    operator: RankingFilterOperator,
+    value: number,
+    attributesOrRefs?: (IAttribute | ObjRefInScope | string)[],
+): IRankingFilter {
```

Comment:
the attributesOrRefs seem 'slapped' onto this factory. check out the newMeasureValueFilter function overloads. they can be used to get rid of optionals on the main methods and achieve nicer typescript API:

`function newRankingFilter(
    measureOrRef: IMeasure | ObjRefInScope | string,
    attributesOrRefs: Array<IAttribute | ObjRefInScope | string>,
    operator: string,
    value: number
): any;

function newRankingFilter(
    measureOrRef: IMeasure | ObjRefInScope | string,
    operator: string,
    value: number
): any;

function newRankingFilter(
    measureOrRef: IMeasure | ObjRefInScope | string,
    attributesOrRefsOrOperator: Array<IAttribute | ObjRefInScope | string> | string,
    operatorOrValue: string | number,
    valueOrNothing?: number
) {
    if (typeof attributesOrRefsOrOperator === "string" && typeof operatorOrValue === "number") {
        // without attributes
        return {}
    } else if (typeof operatorOrValue === "string" && valueOrNothing) {
        // with attributes
        return {}
    } 
    
    throw new InvariantError();
}

function test() {
    newRankingFilter(ReferenceLdm.Won, "op", 10);
    newRankingFilter(ReferenceLdm.Won, [ReferenceLdm.Product.Name], "op", 10)
}`

(when compiled, tsc emits first 2 funs into typings, the third one is the actual implementation and is not in typings; it is in JS code only)